src/components/OrdersWidget/useLowBalance.ts

```diff
@@ -0,0 +1,42 @@
+import { useMemo, useCallback } from 'react'
+import useSafeState from 'hooks/useSafeState'
+
+interface Result {
+  isLowBalance: boolean
+  updateLowBalanceFactory: (id: number) => (isLow: boolean, remove?: true) => void
+}
+
+export function useLowBalance(): Result {
+  const [lowBalance, setLowBalance] = useSafeState<Set<number>>(new Set())
+
+  const updateLowBalanceFactory = useCallback(
+    (id: number) => (isLow: boolean, remove?: true): void => {
+      if (remove || !isLow) {
+        setLowBalance(curr => {
```

Comment:
it looks like this can be factorized a bit, the two branches are super similar and makes it a bit more difficult to understand the difference.

Also, I'm a bit lost with what this does. So maybe some comment. Is this `id` an order id? or is it `orderId + owner`?

Reply:
id == orderId

This useCallback returns a function to be used on each OrderRow to update the state when that order has a low balance.
The set holds ids for all orders with low balance.
If there's anything in the set, the low balance indicator on the top is displayed.

Will make it more verbose.